aurora/src/test/java/com/market/banica/aurora/AuroraComponentIT.java

```diff
@@ -0,0 +1,853 @@
+        }
+
+        ArrayList<Aurora.AuroraResponse> receivedResponses = new ArrayList<>();
+        CountDownLatch latch = new CountDownLatch(1);
+
+        //Act
+        AuroraServiceGrpc
+                .newStub(senderChannel)
+                .subscribe(request, new StreamObserver<Aurora.AuroraResponse>() {
+                    @Override
+                    public void onNext(Aurora.AuroraResponse response) {
+                        receivedResponses.add(response);
+                    }
+
+                    @Override
+                    public void onError(Throwable throwable) {
+                        latch.countDown();
+                    }
+
+                    @Override
+                    public void onCompleted() {
+                        latch.countDown();
+                    }
+                });
+
+        //Assert
+        latch.await();
+        List<TickResponse> unpackedResponses = receivedResponses.stream()
+                .map(response -> {
+                    try {
+                        return response.getMessage().unpack(TickResponse.class);
+                    } catch (InvalidProtocolBufferException e) {
+                        e.printStackTrace();
+                    }
+                    return null;
+                })
+                .collect(Collectors.toList());
+        assertEquals(expectedResponses, unpackedResponses);
+        channelManager.removeChannel("market");
+        publishers.deletePublisher("market");
+    }
+
+    @Test
+    void subscribeToOrderbook_Should_ForwardToReceiverErrorDueNoMapping() throws IOException, InterruptedException {
+        //Arrange
+        publishers.addPublisher("orderbook");
+        createFakeOrderBookServer();
+        createFakeSender();
+
+        grpcCleanup.register(senderChannel);
+        grpcCleanup.register(receiverChannel);
+
+        receiverChannel.getState(true);
+        channelManager.addChannel("orderbook", receiverChannel);
+
+        Aurora.AuroraRequest request = Aurora.AuroraRequest.newBuilder().setTopic("orderbook/something").build();
+
+        final boolean[] inError = {false};
+
+        CountDownLatch latch = new CountDownLatch(1);
+
+
+        //Act
+        AuroraServiceGrpc
+                .newStub(senderChannel)
+                .subscribe(request, new StreamObserver<Aurora.AuroraResponse>() {
+                    @Override
+                    public void onNext(Aurora.AuroraResponse response) {
+
+                    }
+
+                    @Override
+                    public void onError(Throwable throwable) {
+                        inError[0] = true;
+                        latch.countDown();
+                    }
+
+                    @Override
+                    public void onCompleted() {
+                        latch.countDown();
+                    }
+                });
+
+
+        //Assert
+        latch.await();
+        assertTrue(inError[0]);
+        channelManager.removeChannel("orderbook");
```

Comment:
Could you move all .removeChannel and .deletePublisher methods in tearDown method

Reply:
different channels and publishers are being added in different tests in some they are : channelManager.removeChannel("orderbook");
publishers.deletePublisher("orderbook");
in others: 
channelManager.removeChannel("market");
publishers.deletePublisher("market");

I can create a new field of type String which contains the current publisher's name and update it on every test and in the tearDown method I can do: 
channelManager.removeChannel(currentPublisher);
publishers.deletePublisher(currentPublisher);

Reply:
fixed